eclass/kernel-install.eclass

```diff
@@ -38,13 +52,115 @@ esac
+		["dev-libs/libgpg-error"]="GPL-2 LGPL-2.1"
+		["dev-libs/libpcre2"]="BSD"
+		["dev-libs/libtasn1"]="LGPL-2.1+"
+		["dev-libs/lzo"]="GPL-2"
+		["dev-libs/nspr"]="|| ( MPL-2.0 GPL-2 LGPL-2.1 )"
+		["dev-libs/nss"]="|| ( MPL-2.0 GPL-2 LGPL-2.1 )"
+		["dev-libs/oniguruma"]="BSD-2"
+		["dev-libs/openssl"]="Apache-2.0"
+		["media-libs/libpng"]="libpng2"
+		["net-dns/c-ares"]="MIT ISC"
+		["net-fs/cifs-utils"]="GPL-3"
+		["net-fs/nfs-utils"]="GPL-2"
+		["net-fs/samba"]="GPL-3"
+		["net-libs/libmnl"]="LGPL-2.1"
+		["net-libs/libndp"]="LGPL-2.1+"
+		["net-libs/libtirpc"]="BSD BSD-2 BSD-4 LGPL-2.1+"
+		["net-misc/curl"]="BSD curl ISC"
+		["net-misc/dhcp"]="MPL-2.0 BSD SSLeay GPL-2"
+		["net-misc/iputils"]="BSD"
+		["net-misc/networkmanager[iwd]"]="GPL-2+ LGPL-2.1+"
+		["net-nds/rpcbind"]="BSD"
+		["net-wireless/iwd"]="GPL-2"
+		["sys-apps/acl"]="LGPL-2.1"
+		["sys-apps/attr"]="LGPL-2.1"
+		["sys-apps/coreutils"]="GPL-3+"
+		["sys-apps/dbus"]="|| ( AFL-2.1 GPL-2 )"
+		["sys-apps/gawk"]="GPL-3+"
+		["sys-apps/gentoo-systemd-integration"]="BSD"
+		["sys-apps/grep"]="GPL-3+"
+		["sys-apps/hwdata"]="GPL-2+"
+		["sys-apps/iproute2"]="GPL-2"
+		["sys-apps/kbd"]="GPL-2"
+		["sys-apps/keyutils"]="GPL-2 LGPL-2.1"
+		["sys-apps/kmod"]="LGPL-2"
+		["sys-apps/less"]="|| ( GPL-3 BSD-2 )"
+		["sys-apps/net-tools"]="GPL-2"
+		["sys-apps/nvme-cli"]="GPL-2 GPL-2+"
+		["sys-apps/pcsc-lite"]="BSD ISC MIT GPL-3+ GPL-2"
+		["sys-apps/rng-tools"]="GPL-2"
+		["sys-apps/sed"]="GPL-3+"
+		["sys-apps/systemd[boot(-),cryptsetup,pkcs11,tpm,ukify(-)]"]="GPL-2 LGPL-2.1 MIT public-domain "
+		["sys-block/nbd"]="GPL-2"
+		["sys-block/open-iscsi"]="LGPL-3+ GPL-2+"
+		["sys-boot/plymouth"]="GPL-2"
+		["sys-fs/btrfs-progs"]="GPL-2"
+		["sys-fs/cryptsetup"]="GPL-2+"
+		["sys-fs/dmraid"]="GPL-2"
+		["sys-fs/e2fsprogs"]="GPL-2 BSD"
+		["sys-fs/lvm2[lvm]"]="GPL-2"
+		["sys-fs/mdadm"]="GPL-2"
+		["sys-fs/multipath-tools"]="GPL-2"
+		["sys-kernel/dracut"]="GPL-2"
+		["sys-kernel/linux-firmware[redistributable,-unknown-license]"]="BSD-2 BSD BSD-4 GPL-2 GPL-2+ GPL-3 MIT || ( MPL-1.1 GPL-2 ) linux-fw-redistributable ISC MIT"
+		["sys-libs/libcap"]="GPL-2 BSD"
+		["sys-libs/libcap-ng"]="LGPL-2.1"
+		["sys-libs/libnvme"]="LGPL-2.1+"
+		["sys-libs/libseccomp"]="LGPL-2.1"
+		["sys-libs/libxcrypt"]="LGPL-2.1+ public-domain BSD BSD-2"
+		["sys-libs/ncurses"]="MIT"
+		["sys-libs/pam"]="|| ( BSD GPL-2 )"
+		["sys-libs/readline"]="GPL-3+"
+		["sys-libs/zlib"]="ZLIB"
+		["sys-process/procps"]="GPL-2+ LGPL-2+ LGPL-2.1+"
+		["x11-libs/libdrm"]="MIT"
+		["amd64? ( sys-firmware/intel-microcode )"]="amd64? ( intel-ucode )"
+		["x86? ( sys-firmware/intel-microcode )"]="x86? ( intel-ucode )"
+	)
+	LICENSE+="
+		generic-uki? ( ${INITRD_PACKAGES[@]} )
+	"
+fi
+
 SLOT="${PV}"
 IUSE="+initramfs test"
 RESTRICT+="
 	!test? ( test )
 	test? ( userpriv )
 	arm? ( test )
 "
+if [[ ${KERNEL_IUSE_GENERIC_UKI} ]]; then
+	IUSE+=" generic-uki"
+	REQUIRED_USE="
+		generic-uki? ( !initramfs )
+	"
```

Comment:
Oh, sorry, I've missed that `!` before. This is a no-go, and I'm pretty sure that not having `USE=initramfs` forced on this kernel is my mistake. You should update the existing code so that `use generic-uki` takes precedence over `use initramfs`.

Reply:
No problem, this is easily fixed since nowadays the only thing `USE=initramfs` really does is pull in the dracut installkernel plugin.